examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt

```diff
@@ -0,0 +1,265 @@
+                when (intent?.action) {
+                    BluetoothAdapter.ACTION_STATE_CHANGED -> {
+                        val bluetoothState = intent.getIntExtra(
+                                BluetoothAdapter.EXTRA_STATE,
+                                -1
+                        )
+                        when (bluetoothState) {
+                            BluetoothAdapter.STATE_ON -> enableBleServices()
+                            BluetoothAdapter.STATE_OFF -> disableBleServices()
+                        }
+                    }
+                    BluetoothDevice.ACTION_BOND_STATE_CHANGED -> {
+                        val device =
+                                intent.getParcelableExtra<BluetoothDevice>(BluetoothDevice.EXTRA_DEVICE)
+                        Log.d(
+                                TAG,
+                                "Bond state changed for device ${device?.address}: ${device?.bondState}"
+                        )
+                        when (device?.bondState) {
+                            BluetoothDevice.BOND_BONDED -> addDevice(device)
+                            BluetoothDevice.BOND_NONE -> removeDevice(device)
+                        }
+                    }
+
+                }
+            }
+        }
+        registerReceiver(bluetoothObserver, IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED))
+        registerReceiver(bluetoothObserver, IntentFilter(BluetoothDevice.ACTION_BOND_STATE_CHANGED))
+
+        // Startup BLE if we have it
+
+        val bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+        if (bluetoothManager.adapter?.isEnabled == true) enableBleServices()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        unregisterReceiver(bluetoothObserver)
+        disableBleServices()
+    }
+
+    override fun onBind(intent: Intent?): IBinder? =
+            when (intent?.action) {
+                DATA_PLANE_ACTION -> {
+                    DataPlane()
+                }
+                else -> null
+            }
+
+    override fun onUnbind(intent: Intent?): Boolean =
+            when (intent?.action) {
+                DATA_PLANE_ACTION -> {
+                    myCharacteristicChangedChannel = null
+                    true
+                }
+                else -> false
+            }
+
+    /**
+     * A binding to be used to interact with data of the service
+     */
+    inner class DataPlane : Binder() {
+        fun setMyCharacteristicChangedChannel(sendChannel: SendChannel<String>) {
+            myCharacteristicChangedChannel = sendChannel
+        }
+    }
+
+    private fun enableBleServices() {
+        val bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+        if (bluetoothManager.adapter?.isEnabled == true) {
+            Log.i(TAG, "Enabling BLE services")
+            bluetoothManager.adapter.bondedDevices.forEach { device -> addDevice(device) }
```

Comment:
If I bond a device via system settings, then should I expect that it should be connectable via BLE i.e. bonding information is the same no matter whether the pairing was done for BT classic or BLE on the proviso that the peripheral device supports BLE?

Reply:
Yes. Bonding in Settings should I think be preferred way, add them the app doesn't need location permission and location service in order to scan for Ble devices.